motionplan/collision.go

```diff
@@ -55,169 +54,110 @@ type collisionEntity struct {
 	geometry spatial.Geometry
 }
 
-// CollisionEntities defines an interface for a set of collisionEntities that can be treated as a single batch.
-type CollisionEntities interface {
-	count() int
-	entityFromIndex(int) *collisionEntity
-	indexFromName(string) int
-	checkCollision(*collisionEntity, *collisionEntity, bool) (float64, error)
-	reportCollisions([]float64) []int
-}
-
-// ObjectCollisionEntities is an implementation of CollisionEntities for entities that occupy physical space and should not be intersected
+// collisionEntities is an implementation of CollisionEntities for entities that occupy physical space and should not be intersected
 // it is exported because the key CollisionEntities in a CollisionSystem must be of this type.
-type ObjectCollisionEntities struct {
+type collisionEntities struct {
 	entities []*collisionEntity
-	indices  map[string]int
+	indices  *map[string]int
 }
 
-// NewObjectCollisionEntities is a constructor for ObjectCollisionEntities, an exported implementation of CollisionEntities.
-func NewObjectCollisionEntities(geometries map[string]spatial.Geometry) (*ObjectCollisionEntities, error) {
+func newCollisionEntities(geometries map[string]spatial.Geometry) (*collisionEntities, error) {
```

Comment:
It feels weird to still have `collisionEntities` as a general concept.

It's functionally a somewhat-obfuscated re-implementation of the same map that we pass in.

It seems like everything this is doing, here and below, can be done using the original map[string]geometry and using the strings for identification rather than reducing them to index numbers.